app/Http/Controllers/Admin/RouteController.php

```diff
@@ -98,7 +97,31 @@ public function edit($id)
      */
     public function update(RouteRequest $request, $id)
     {
-        Route::findOrFail($id)->update($request->all());
+        $allRequest = $request->all();
+        DB::transaction(function () use ($allRequest, $id) {
+            Route::findOrFail($id)->update($allRequest);
+            Direction::where('route_id', '=', $id)->delete();
+            for ($i=0; $i<count($allRequest['stop_id_forward']); $i++) {
```

Comment:
these code is duplicate on store and update , pls follow `DRY`  rules. Separate to another function and just call